components/brave_rewards/browser/rewards_service_browsertest.cc

```diff
@@ -1716,9 +1729,54 @@ IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest,
       ui_test_utils::BROWSER_TEST_WAIT_FOR_NAVIGATION);
 
   // Ensure that Twitter tips injection is not active
-  EXPECT_FALSE(IsTwitterTipsInjected());
+  EXPECT_FALSE(IsMediaTipsInjected());
 }
 
+// Brave tip icon is injected when visiting Reddit
+IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest, RedditTipsInjectedOnReddit) {
+  // Enable Rewards
+  EnableRewards();
+
+  // Navigate to Reddit in a new tab
+  GURL url = https_server()->GetURL("reddit.com", "/reddit");
+  ui_test_utils::NavigateToURLWithDisposition(
+      browser(), url, WindowOpenDisposition::NEW_FOREGROUND_TAB,
+      ui_test_utils::BROWSER_TEST_WAIT_FOR_NAVIGATION);
+
+  // Ensure that Reddit Tips injection is active
+  EXPECT_TRUE(IsMediaTipsInjected());
+}
+
+// Brave tip icon is injected when visiting Reddit
+IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest,
+                       RedditTipsNotInjectedWhenRewardsDisabled) {
+  // Navigate to Reddit in a new tab
+  GURL url = https_server()->GetURL("reddit.com", "/reddit");
+  ui_test_utils::NavigateToURLWithDisposition(
+      browser(), url, WindowOpenDisposition::NEW_FOREGROUND_TAB,
+      ui_test_utils::BROWSER_TEST_WAIT_FOR_NAVIGATION);
+
+  // Ensure that Reddit Tips injection is active
+  EXPECT_FALSE(IsMediaTipsInjected());
+}
+
+// Brave tip icon is injected when visiting Reddit
+IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest,
+                       RedditTipsNotInjectedOnNonReddit) {
+  // Enable Rewards
+  EnableRewards();
+
+  // Navigate to Reddit in a new tab
+  GURL url = https_server()->GetURL("brave.com", "/reddit");
+  ui_test_utils::NavigateToURLWithDisposition(
+      browser(), url, WindowOpenDisposition::NEW_FOREGROUND_TAB,
+      ui_test_utils::BROWSER_TEST_WAIT_FOR_NAVIGATION);
+
+  // Ensure that Reddit Tips injection is active
```

Comment:
Looks like this comment doesn't match the test.

Reply:
Fixed